### PR TITLE
[SYCL-MLIR] Avoid evaluating expressions only used on supressed warnings

### PR DESCRIPTION
**Pull request by victor-eds**
Define CGEIST_WARNING macro to avoid evaluating not needed expressions if -w is passed.

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

File: polygeist/tools/cgeist/Lib/CGExpr.cc [hunk 1]
```diff
@@ -1214,10 +1215,12 @@ ValueCategory MLIRScanner::VisitAtomicExpr(clang::AtomicExpr *BO) {
...
-  for (auto &Child : E->children()) {
-    mlirclang::warning() << "cleanup not handled for: ";
-    Child->dump(mlirclang::warning(), Glob.getCGM().getContext());
-  }
+  CGEIST_WARNING({
+    for (auto &Child : E->children()) {
+      mlirclang::warning() << "cleanup not handled for: ";
+      Child->dump(mlirclang::warning(), Glob.getCGM().getContext());
+    }
+  });
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Lib/CGExpr.cc:1223]: We avoid evaluating a whole loop here.

File: polygeist/tools/cgeist/Lib/utils.cc [hunk 2]
```diff
@@ -89,10 +87,6 @@ gpu::GPUModuleOp getDeviceModule(ModuleOp Module) {
       Module.lookupSymbol(MLIRASTConsumer::DeviceModuleName));
 }
 
-llvm::raw_ostream &warning() {
-  if (SuppressWarnings)
-    return llvm::nulls();
-  return llvm::WithColor::warning();
-}
+llvm::raw_ostream &warning() { return llvm::WithColor::warning(); }
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Lib/utils.cc:90]: Always return `warning()` now.

Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Lib/utils.cc:90]: Should we directly use `llvm::WithColor::warning()` and remove `mlirclang::warning()`? so is clear the intent.

Comment (victor-eds) [reply polygeist/tools/cgeist/Lib/utils.cc]: Agree. I though of doing that, but wanted to discuss beforehand. Done.

**Review by whitneywhtsang — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Pull request merged**